lib/agent.js

```diff
@@ -48,22 +101,13 @@ function dispatchFromAgent (requestType) {
       throw new InvalidArgumentError('unsupported opts.path')
     }
 
-    // if url is a string transform into URL inst. otherwise use as is
-    url = typeof url === 'string' ? new URL(url) : url
-    // ensure it atleast has a non-empty string origin
-    if (!url || typeof url.origin !== 'string' || url.origin === '') {
-      throw new InvalidArgumentError('Argument url.origin must be a non-empty string')
-    }
-
     const client = agent.get(url.origin)
 
     if (client && typeof client[requestType] !== 'function') {
       throw new InvalidReturnValueError(`Client returned from Agent.get() does not implement method ${requestType}`)
     }
 
-    const path = url.path || `${url.pathname || '/'}${url.search || ''}`
-
-    return client[requestType]({ ...opts, method, path }, ...additionalArgs)
+    return client[requestType]({ ...opts, method, path: `${url.pathname || '/'}${url.search || ''}` }, ...additionalArgs)
```

Comment:
actually url can't contain pathname nor search, it's just origin URL.

Reply:
I think we're both confused :D

Reply:
Maybe just skip parseOrigin here for now and do this part like you did before.

Reply:
Okay cool